adapters/repos/db/inverted/new_prop_length_tracker.go

```diff
@@ -0,0 +1,292 @@
+					}
+
+					data.SumData[name] = data.SumData[name] + int(value)*int(count)
+					data.CountData[name] = data.CountData[name] + int(count)
+
+				}
+			}
+			t.data = data
+			t.Flush(true)
+			plt.Close()
+			plt.Drop()
+			t.Flush(false)
+		}
+	}
+	t.data = data
+
+	return t, nil
+}
+
+// Path to the file on disk
+func (t *JsonPropertyLengthTracker) FileName() string {
+	return t.path
+}
+
+// Adds a new value to the tracker
+func (t *JsonPropertyLengthTracker) TrackProperty(propName string, value float32) error {
+	t.Lock()
+	defer t.Unlock()
+
+	t.data.SumData[propName] = t.data.SumData[propName] + int(value)
+	t.data.CountData[propName] = t.data.CountData[propName] + 1
+
+	bucketId := t.bucketFromValue(value)
+	if _, ok := t.data.BucketedData[propName]; ok {
+		t.data.BucketedData[propName][int(bucketId)] = t.data.BucketedData[propName][int(bucketId)] + 1
+	} else {
+
+		t.data.BucketedData[propName] = make(map[int]int, 64)
+		t.data.BucketedData[propName][int(bucketId)] = 1
+	}
+
+	return nil
+}
+
+// Removes a value from the tracker
+func (t *JsonPropertyLengthTracker) UnTrackProperty(propName string, value float32) error {
+	t.Lock()
+	defer t.Unlock()
+
+	t.data.SumData[propName] = t.data.SumData[propName] - int(value)
+	t.data.CountData[propName] = t.data.CountData[propName] - 1
+
+	bucketId := t.bucketFromValue(value)
+	if _, ok := t.data.BucketedData[propName]; ok {
+		t.data.BucketedData[propName][int(bucketId)] = t.data.BucketedData[propName][int(bucketId)] - 1
+	} else {
+		return errors.New("property not found")
+	}
+
+	return nil
+}
+
+// Returns the bucket that the given value belongs to
+func (t *JsonPropertyLengthTracker) bucketFromValue(value float32) int {
+	if t.UnlimitedBuckets {
+		return int(value)
+	}
+	if value <= 5.00 {
+		return int(value) - 1
+	}
+
+	bucket := int(math.Log(float64(value)/4.0)/math.Log(1.25) + 4)
+	if bucket > 63 {
+		return 64
+	}
+	return int(bucket)
+}
+
+// Returns the value that the given bucket represents
+//
+//nolint:all
+func (t *JsonPropertyLengthTracker) valueFromBucket(bucket int) float32 {
+	if t.UnlimitedBuckets {
+		return float32(bucket)
+	}
+	if bucket <= 5 {
+		return float32(bucket + 1)
+	}
+
+	if bucket > 63 {
+		return -1
+	}
+
+	return float32(4 * math.Pow(1.25, float64(bucket)-3.5))
+}
+
+// Returns the average length of the given property
+func (t *JsonPropertyLengthTracker) PropertyMean(propName string) (float32, error) {
+	t.Lock()
+	defer t.Unlock()
+	sum, ok := t.data.SumData[propName]
+	if !ok {
+		return 0, nil
+	}
+	count, ok := t.data.CountData[propName]
+	if !ok {
+		return 0, nil
+	}
+
+	return float32(sum) / float32(count), nil
+}
+
+// returns totalPropertyLength, totalCount, average propertyLength = sum / totalCount, total propertylength, totalCount, error
+func (t *JsonPropertyLengthTracker) PropertyTally(propName string) (int, int, float64, error) {
+	t.Lock()
+	defer t.Unlock()
+	sum, ok := t.data.SumData[propName]
+	if !ok {
+		return 0, 0, 0, nil
```

Comment:
return an error here?

Reply:
Backwards compatibility :(

Reply:
ah bummer, a short comment would be helpful for the next one.

What does it mean if a prop is not found in the SumData? That no object was added that has that property defined?

Reply:
Yes, the props only get created if there is data for them